File: NeetCode-Questions/1-Arrays & Hashing/Q20.java
import java.util.Stack;

public class Q20 {
    public static void main(String[] args) {
        String s = "()[]{}";

        Stack<Character> stack = new Stack<>();

        for (char bracket : s.toCharArray()) {

            if (bracket == '(') {
                stack.push(')');
            } else if (bracket == '[') {
                stack.push(']');
            } else if (bracket == '{') {
                stack.push('}');
            } else {
                if (stack.isEmpty() || stack.peek() != bracket) {
                    // return false;
                } else {
                    stack.pop();
                }
            }
        }

        System.out.println(stack.size());

    }
}

File: NeetCode-Questions/4- stack/Q20.py
s = "]"

stack = []

for i in range(len(s)):
    if s[i] == "(":
        stack.append(")")
    elif s[i] == "{":
        stack.append("}")
    elif s[i] == "[":
        stack.append("]")
    else:
        if len(stack) == 0 or stack[-1] != s[i]:
              return False
        else:
            stack.pop()
print(stack)